src/Dropzone/index.jsx

```diff
@@ -0,0 +1,282 @@
+  });
+
+  const renderContent = () => {
+    if (isMultipleDragged) {
+      return <DragError message={multipleDraggedMsg} />;
+    }
+
+    if (errors.length > 0) {
+      return (
+        <>
+          <GenericError errorMsgs={errors} />
+          <DefaultContent />
+        </>
+      );
+    }
+
+    if (progress && progress !== 100) {
+      return (
+        <UploadProgress
+          variant={progressVariant}
+          percent={progress}
+          name={fileName}
+          onCancel={handleUploadCancel}
+        />
+      );
+    }
+
+    return inputComponent;
+  };
+
+  return (
+    <section>
+      <div
+        {...getRootProps({
+          className: classNames('pgn__dropzone', {
+            className,
+            'pgn__dropzone-validation-error': isMultipleDragged || errors.length > 0 || isDragReject,
+            'pgn__dropzone-active': isDragActive && !isDragReject,
+          }),
+        })}
+      >
+        <input {...getInputProps()} />
+        <div className="d-flex flex-column justify-content-around align-items-center w-100">
+          {renderContent()}
+        </div>
+      </div>
+    </section>
+  );
+};
+
+Dropzone.defaultProps = {
+  className: undefined,
+  accept: undefined,
+  maxSize: Infinity,
+  minSize: 0,
+  uploadCallbacks: {
+    onUploadProgress: () => {},
+    onUploadSuccess: () => {},
+    onUploadCancel: () => {},
+    onUploadError: () => {},
+  },
+  errorMessages: {
+    invalidType: undefined,
+    invalidSize: undefined,
+    multipleDragged: 'Only one upload permitted.',
+    uploadError: 'A problem occured while uploading your file. Please try again.',
+  },
+  progressVariant: 'spinner',
+  validator: undefined,
+  inputComponent: <DefaultContent />,
+};
+
+Dropzone.propTypes = {
+  /** Specifies class name to append to the base element. */
+  className: PropTypes.string,
+  /**
+   * Set accepted file types.
+   * This should be an object with the keys set to the
+   * [MIME type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types)
+   * and the values to an array of file extensions.
+   */
+  accept: PropTypes.objectOf(PropTypes.arrayOf(PropTypes.string)),
+  /** Maximum file size (in bytes). */
+  maxSize: PropTypes.number,
+  /** Minimum file size (in bytes). */
+  minSize: PropTypes.number,
+  /**
+   * An object containing callbacks for upload events, currently four events are supported:
+   * 1) onUploadProgress - A callback fired each time an upload progress event happens,
+   * receives (percentageUploaded, progressEvent) as arguments.
+   * 2) onUploadSuccess - A callback fired upon successful upload, receives Response object as a single argument.
+   * 3) onUploadCancel - A callback fired upon cancelling an upload, receives no arguments.
+   * 4) onUploadError - A callback fired if upload was not successful, receives Error object as a single argument.
+   */
+  uploadCallbacks: PropTypes.shape({
```

Comment:
yeah, I think your approach is better here, I recently read some article that said too many props is kinda an anti-pattern for React, and grouping some of them into objects is one way to avoid it. Tried to do it here, but it didn't feel right, I guess for component library it's OK to have many props 😄 

Although for error messages I think grouping is fine, what do you think?